core/src/main/scala/kafka/server/ReplicaFetcherThread.scala

```diff
@@ -132,9 +139,22 @@ class ReplicaFetcherThread(name: String,
 
     brokerTopicStats.updateReplicationBytesIn(records.sizeInBytes)
 
+    val highWatermarkChanged = log.maybeUpdateHighWatermark(partitionData.highWatermark)
+    if (highWatermarkChanged) {
+      logAppendInfo.foreach { _ => partitionsWithNewHighWatermark += topicPartition }
```

Comment:
nit: I think that we don't have to do `logAppendInfo.foreach` here anymore, right?